lib/manager/components/LabelEditor.js

```diff
@@ -0,0 +1,234 @@
+  }
+
+  _onSaveSettings = () => {
+    const {labelIsNew, newLabel: label} = this.state
+    const {createLabel, updateLabel} = this.props
+
+    if (labelIsNew) {
+      // Insert project ID
+      label.projectId = this.props.projectId
+      createLabel(label)
+    } else {
+      updateLabel(label)
+    }
+
+    // This passed method will close the editor
+    this.props.onDone && this.props.onDone()
+  }
+
+  render () {
+    const { newLabel, validation } = this.state
+    const { user } = this.props
+
+    return (
+      <Form>
+        <Panel>
+          <ListGroup fill>
+            <ListGroupItem>
+              <FormGroup validationState={validationState(validation.name)}>
+                <ControlLabel>Name</ControlLabel>
+                <FormControl
+                  value={newLabel.name}
+                  name={'name'}
+                  onChange={this._onFormChange}
+                />
+                <FormControl.Feedback />
+                <HelpBlock>Required.</HelpBlock>
+              </FormGroup>
+              <FormGroup>
+                <ControlLabel>Description</ControlLabel>
+                <FormControl
+                  value={newLabel.description}
+                  name={'description'}
+                  onChange={(e) => this._onFormChange(e)}
+                />
+                <FormControl.Feedback />
+              </FormGroup>
+            </ListGroupItem>
+            <ListGroupItem>
+              <Row>
+                <Col xs={3}>
+                  <FormGroup>
+                    <ControlLabel>Color</ControlLabel>
+                    <FormControl
+                      value={newLabel.color}
+                      name={'color'}
+                      type='color'
+                      onChange={this._onFormChange}
+                    />
+                    <FormControl.Feedback />
+                  </FormGroup>
+                </Col>
+                {this.userIsAdmin(user) && (
+                  <Col xs={9}>
+                    <Checkbox
+                      checked={newLabel.adminOnly}
+                      name='adminOnly'
+                      type='checkbox'
+                      onChange={(e) => this._onCheckboxChange(e)}
+                      style={{margin: 0}}
+                    >
+                      <strong>Visible to admins only?</strong>
+                    </Checkbox>
+                    <div style={{marginTop: 5}}>
+                      If checked, this label will only be visible to project and application admins. Users with feed-specific priveleges will not see this label applied to any feed sources.
+                    </div>
+                  </Col>
+                )}
+              </Row>
+            </ListGroupItem>
+          </ListGroup>
+        </Panel>
+        <Modal.Footer>
+          {/* Cancel button */}
+          <Button onClick={this.props.onDone} style={{ marginRight: 10 }}>
+              Cancel
```

Comment:
Fix indent

Reply:
This one was done by prettier but I've put it all one one line for now